### PR TITLE
Clean up for new Azure Publish layout

### PR DESCRIPTION
**Pull request by eerhardt**
This is the 2nd part to our new Azure Publish layout.

- Fixing dotnet-install to consume the new layout.
- Fixing the README to point to the correct links.
- Also, fixing a minor issue with the URLs in "FinalizeBuild"

@brthor @livarcocc @piotrpMSFT @sokket

### CODE REVIEW

**Comment by livarcocc** [comment]
:shipit:

**Comment by brthor** [comment]
:shipit:

**Comment by brthor** [comment]
:shipit:

**Comment by brthor** [comment]
:shipit:

**Comment by brthor** [comment]
:shipit:

File: scripts/obtain/dotnet-install.ps1 [hunk 1]
```diff
@@ -50,7 +50,7 @@
 #>
 [cmdletbinding()]
 param(
-   [string]$Channel="preview",
+   [string]$Channel="rel-1.0.0",
```
Comment (andschwa) [inline scripts/obtain/dotnet-install.ps1]: Oh, that's why our builds broke :tongue:

Comment (brthor) [inline scripts/obtain/dotnet-install.ps1]: If you are pulling preview2 bits I recommend you use the install script from the preview2 channel to keep things working there

Comment (eerhardt) [inline scripts/obtain/dotnet-install.ps1]: Yes, this is kind of the problem with 'dotnet-install' today, it is tied to the branch we are developing in.

The good news is that this "Publish Refactor" work that I'm doing is setting us up for success long term. After this, our publish output will be in the layout that we can maintain going forward.  So then the 'dotnet-install' script won't have to change anymore, once everything is in this new layout.  (I'm looking at you, `dotnet/core-setup`)

Comment (andschwa) [inline scripts/obtain/dotnet-install.ps1]: We just switched to `rel-1.0.0` channel, all is good now :+1:

**Pull request merged**